### PR TITLE
feat: better transferring

### PR DESCRIPTION
**Pull request by okikio**
There are a couple problems with the way `transferables` work. You currently don't check for runtime support, and neither do you check for `transferables` in complex objects. I'm using  [@okikio/transferables](https://jsr.io/@okikio/transferables) to cover the `transferables` usecases. Also, for the `isTransferables` you're currently using `getGlobalVar` which is a potential security risk as it calls `Function("return ...")` if someone passes in custom code, you can potentially have XSS attacks.

### CODE REVIEW

File: src/rpc/rpc.ts [hunk 1]
```diff
@@ -9,13 +10,28 @@ import type {
...
 const RPC_AGENT = Symbol("rpcAgent");
 
+// Support flags for features like streams and channels, 
+// we want to confirm if streams and channels are transferable
+// transferables aren't supported the same way across all browsers and runtimes,
+// in some runtimes it's more like partial support
+let supportFlags: Awaited<ReturnType<typeof isSupported>> = { streams: false, channel: false };
+(async () => {
+  await new Promise<void>((resolve) => {
+    isSupported()
```
Comment (mys1024) [inline src/rpc/rpc.ts]: image
These outputs are unnecessary. It seems that they come from [this line](https://jsr.io/@okikio/transferables/1.0.1/src/mod.ts#L306).
Currently, this project's `deno task test` will crash on Deno v1.41.2, you can use Deno v1.41.0 instead for testing.

File: src/rpc/rpc.ts [hunk 2]
```diff
@@ -9,13 +10,28 @@ import type {
...
+      .then((res) => (supportFlags = res, resolve()));
+
+    // Just in-case some runtime takes longer that we want to confirm support for streams and/or channels
+    // The `isSupported` function is supposed to be fairly fast, but just in-case it isn't we set a 3sec cutoff
+    setTimeout(() => resolve(), 3000);
+  });
+})();
```
Comment (mys1024) [inline src/rpc/rpc.ts]:
```typescript
(async () => {
  await new Promise<void>((resolve) => {
    isSupported()
      .then((res) => (supportFlags = res, resolve()));

    // Just in-case some runtime takes longer that we want to confirm support for streams and/or channels
    // The `isSupported` function is supposed to be fairly fast, but just in-case it isn't we set a 3sec cutoff
    setTimeout(() => resolve(), 3000);
  });
})();
```

What's the purpose of the cutoff? This IIFE seems to behave like the following simple line:

```typescript
isSupported().then((res) => supportFlags = res);
```

**Review by mys1024 — CHANGES_REQUESTED** [review]

File: src/rpc/rpc.ts (hunk 2 above)
Comment (okikio) [reply src/rpc/rpc.ts]: Done. Sorry for the late response

File: src/rpc/rpc.ts (hunk 1 above)
Comment (okikio) [reply src/rpc/rpc.ts]: Done

**Comment by okikio** [comment]
Sorry, this week was a little busy for me

**Comment by mys1024** [comment]
@okikio No need to be sorry, thanks for your contribution! 🎉

**Review by mys1024 — APPROVED** [review]
LGTM!

**Pull request merged**